cypress/integration/console/devices/import.spec.js

```diff
@@ -80,4 +80,53 @@ describe('End device messaging', () => {
         cy.findByText(/an end device with/, /is already registered/).should('be.visible')
       })
   })
+
+  it('succeeds setting lorawan_version, lorawan_phy_version and frequency_plan_id from fallback values', () => {
+    const devicesFile = 'freqId-version-phy-device.json'
+    const fallbackValues = {
+      lorawan_version: 'MAC_V1_0',
+      frequency_plan_id: '863-870 MHz',
+    }
+    cy.findByLabelText('File format').selectOption('The Things Stack JSON')
+    cy.findByLabelText('File').attachFile(devicesFile)
+    cy.findByLabelText('Frequency plan').selectOption(fallbackValues.frequency_plan_id)
+    cy.findByLabelText('LoRaWAN version').selectOption(fallbackValues.lorawan_version)
+
+    cy.findByRole('button', { name: 'Import end devices' }).click()
+    cy.findByText('Operation finished').should('be.visible')
+    cy.findByText('3 of 3 (100.00% finished)').should('be.visible')
+    cy.findByTestId('notification')
+      .should('be.visible')
+      .findByText('All end devices imported successfully')
+      .should('be.visible')
+    cy.findByRole('button', { name: 'Proceed to end device list' }).click()
+    cy.location('pathname').should(
+      'eq',
+      `${Cypress.config('consoleRootPath')}/applications/${appId}/devices`,
+    )
+    cy.findByTestId('error-notification').should('not.exist')
+    cy.findByRole('rowgroup').within(() => {
+      cy.findAllByRole('row').should('have.length', 7)
```

Comment:
This assertion will make the test dependant on the preceding tests in the spec. It's important to write isolated test specs so that rearranging or modifying other tests will not affect the result of this one. What you could do here:
- Reseed the database before the test(s)
- Do not assert on the count of table rows but specifically look for the names of the imported devices

But I think what is even better, is to only once verify that devices actually exist after the import was prompted as successful and only once check whether failed devices have actually not been created. This could be done in the first two tests of this spec.

Reply:
Additionally, as part of this test, you should also check whether MAC/PHY/FP have actually been set as expected (e.g. the fallback has not overwritten a value already set in the import file) This could be done by navigating to the general settings of an imported device, expanding the network layer form, and verifying the MAC/PHY/FP.